### ISSUE TITLE
Java: Add ArrayTypeAccess C-style predicate

### ISSUE THREAD

**Issue by Marcono1234** [github/codeql#3624]
**Description of the issue**
Would it be possible to add a predicate to `ArrayTypeAccess` which returns whether the type access uses the C-style notation?
C-style array declarations are discouraged and you can severly decrease readability:
```
int[] a, b[];

int[] test()[] {
  return null;
}
```
is equivalent to:
```
int[] a;
int[][] b;

int[][] test() {
  return null;
}
```

See also [JLS 14 §10.2](https://docs.oracle.com/javase/specs/jls/se14/html/jls-10.html#jls-10.2)

**Comment by aschackmull** [comment]
This is unfortunately not information that we directly extract, so it'll be a bit tricky to implement as it would need to involve some careful inspection of the location ranges.  I can definitely see that it would be nice to have, so we may look into this.

**Comment by Marcono1234** [comment]
Yes looks like this is indeed possible using the location ranges, thanks for the hint!
See [this implementation](https://github.com/Marcono1234/codeql-java-queries/blob/master/queries/c-style-array.ql) (though I have not tested it for C-style method return type yet).
